tests/manage/z_cluster/nodes/test_check_pod_status_after_two_nodes_shutdown_recovery.py

```diff
@@ -0,0 +1,162 @@
+        """
+        Restart nodes that are in status NotReady
+        for situations in which the test failed in between
+
+        """
+
+        def finalizer():
+
+            # Validate all nodes are in READY state
+            not_ready_nodes = [
+                n for n in get_node_objs() if n
+                .ocp.get_resource_status(n.name) == constants.NODE_NOT_READY
+            ]
+            log.warning(
+                f"Nodes in NotReady status found: {[n.name for n in not_ready_nodes]}"
+            )
+            if not_ready_nodes:
+                nodes.restart_nodes_by_stop_and_start(not_ready_nodes)
+                wait_for_nodes_status()
+
+            log.info("All nodes are in Ready status")
+
+        request.addfinalizer(finalizer)
+
+    def list_of_nodes_running_pods(self, selector):
+        """
+        The function returns the list of nodes for the given selector
+
+        Args:
+            selector (str): The resource selector to search with
+
+        Returns:
+            list: Pods_running_nodes
+
+        """
+        pod_obj_list = get_all_pods(
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE, selector=[selector]
+        )
+        pods_running_nodes = []
+        for pod in pod_obj_list:
+            pods_running_nodes.append(
+                get_node_running_pod(
+                    namespace=defaults.ROOK_CLUSTER_NAMESPACE, pod_name=pod.name
+                )
+            )
+
+        log.info(f"{selector} running on nodes {pods_running_nodes}")
+        return pods_running_nodes
+
+    @pytest.mark.polarion_id("OCS-1278")
+    @pytest.mark.polarion_id("OCS-2315")
+    def test_node_after_shutdown_and_recovery_worker_node(self, nodes):
+        """
+        Test case to check MDS pods and Provisioner pods not running on same node
+        post shutdown and recovery node
+
+        """
+
+        # Get MDS pods running nodes before shutdown
+
+        self.list_of_nodes_running_pods(
+            selector='rook-ceph-mds'
+        )
+
+        self.list_of_nodes_running_pods(
+            selector='csi-rbdplugin-provisioner'
+        )
+
+        self.list_of_nodes_running_pods(
+            selector='csi-cephfsplugin-provisioner'
+        )
+
+        # Get the node list
+        node = get_typed_nodes(node_type='worker', num_of_nodes=2)
+
+        # Shutdown 2 worker nodes for 10 mins
+        nodes.stop_nodes(nodes=node)
+
+        waiting_time = 600
+        log.info(f"Waiting for {waiting_time} seconds")
+        time.sleep(waiting_time)
+
+        nodes.start_nodes(nodes=node)
+
+        # Validate all nodes are in READY state and up
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=30,
+            delay=15)(
+            wait_for_nodes_status(timeout=1800)
+        )
+
+        # Check the node are Ready state and check cluster is health ok
+        self.sanity_helpers.health_check()
+
```

Comment:
Also check for all ocs pods status and wait till they reaches running state.
You can use ocs_ci.ocs.resources.pod.wait_for_storage_pods

Reply:
done